### PR TITLE
feat(motion): add motion to experimental side nav

### PR DESCRIPTION
**Pull request by tw15egan**
Refs https://github.com/carbon-design-system/carbon-website/issues/838

Adds in motion on expansion/collapse of side nave menu items 

#### Changelog

**New**

- Added some css animations on `side-nav__menu`

#### Testing / Reviewing

Test out the animations work on UI Shell - Fixed Side Nav example

### CODE REVIEW

File: src/components/ui-shell/_side-nav.scss [hunk 1]
```diff
@@ -4,7 +4,7 @@
 // This source code is licensed under the Apache-2.0 license found in the
 // LICENSE file in the root directory of this source tree.
 //
-
+$count: 0;
```
Comment (asudoh) [inline src/components/ui-shell/_side-nav.scss]: Do you think this variable is still needed?

File: src/components/ui-shell/_side-nav.scss [hunk 2]
```diff
@@ -306,12 +307,45 @@
...
 
   .#{$prefix}--side-nav__submenu[aria-expanded='true'] + .#{$prefix}--side-nav__menu[role='menu'] {
-    display: block;
+    transition-timing-function: $carbon--ease-in;
+    max-height: 2000px;
+
+    .bx--side-nav__link {
+      transform: translateY(0);
+      opacity: 1;
+      visibility: visible;
```
Comment (asudoh) [inline src/components/ui-shell/_side-nav.scss]: [`visibility:visible` shows the content even in hidden outer element](https://jakearchibald.com/2014/visible-undoes-hidden/) - Probably `visibility:inherit`?

**Review by asudoh — COMMENTED** [review]
Just a couple of comments - Thanks @tw15egan!

**Comment by tw15egan** [comment]
@asudoh updated!

**Review by shixiedesign — APPROVED** [review]
💯

**Review by asudoh — APPROVED** [review]
Thanks @tw15egan for your quick update!

**Pull request merged**